torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -733,21 +730,50 @@ def call_function(
             UserFunctionVariable,
             NestedUserFunctionVariable,
         )
-        assert type(args[1].realize()) is TensorVariable
 
-        sample_shape = get_fake_value(args[1].as_proxy().node, tx).size()
+        assert only_consist_of(args[1], TensorVariable)
+        xs_proxy = args[1].as_proxy()
+        flat_proxies, xs_spec = pytree.tree_flatten(xs_proxy)
+        xs_example_values = [p.node.meta["example_value"] for p in flat_proxies]
 
-        if len(sample_shape) < 1 or sample_shape[0] == 0:
+        same_first_dim_msg = "map expects all mapped tensors to have the same size in the first dimension."
+        s0 = xs_example_values[0].size()[0]
+        if s0 == 0:
             unimplemented(
-                "map() operator doesn't support scalar or zero-sized tensors during tracing."
+                "map expects all mapped tensors to have non-zero size in the first dimension."
```

Comment:
Can we put this check into a Python function and then have Dynamo call into it? Because we probably want to re-use the map validation function in the actual implementation of map as well

Reply:
That's a good point.